### ISSUE TITLE
Slow loading for second model

### ISSUE THREAD

**Issue by thiriob** [atteneder/glTFast#165]
Hello
I load gltfs this way :
```
var gltf = new GLTFast.GLTFast();
await gltf.Load(myUri);
gltf.InstantiateGltf(new CustomInstantiator());
```
This works perfectly fine but when I load an entire new scene and come back to this loading, it becomes much slower.
The fix I found was to go to GltFast.cs line 115 and remove the static from :
```
static IDeferAgent defaultDeferAgent;
```

This solution is ok for now but doesnt seem ideal.

**Comment by atteneder** [comment]
Hi,

Thanks for reaching out.

> This works perfectly fine but when I load an entire new scene and come back to this loading, it becomes much slower.

Do you mean you load a different glTF within the same Unity scene (on top/after loading the first glTF) ?
Loading the two scenes individually (without loading another one first) is faster?

> The fix I found was to go to GltFast.cs line 115 and remove the static from :
> 
> ```
> static IDeferAgent defaultDeferAgent;
> ```
> 
> This solution is ok for now but doesnt seem ideal.

You're basically not re-using the defaultDeferAgent anymore, which might speed up individual, parallel loading files at the cost of breaking the frame-rate (in theory). But this raises some questions, whether there's a weak spot in the defer system. Would you be able to provide sample files and a script that demonstrate this behaviour?

Regards,
Andreas

**Comment by atteneder** [comment]
Please provide a sample file/project and re-open.

**Comment by atteneder** [comment]
Thanks, will look into it!

**Comment by merulaxr** [comment]
We have also encountered this issue, I've traced it to the accessor switch statement https://github.com/atteneder/glTFast/blob/main/Runtime/Scripts/GltfImport.cs#L2211 and in particular the `await deferAgent.BreakPoint();` that follows on Line 2271. Tested with the built in GltfAsset loader.

**Comment by atteneder** [comment]
Thanks everyone for finding this bug! Especially @merulaxr for providing a fix and nailing down the issue!

The problem was, that the auto-generated default `IDeferAgent` (responsible for delaying loading work to the next frame, to keep the frame rate stable) was destroyed when loading a different scene and this went unnoticed, leading to the slow performance you observed.

I installed a couple of security measurements:

- The auto-generated default `IDeferAgent` is flagged `DontDestroyOnLoad`, so it survives scene loads. This saves resources.
- If it still is destroyed (by force), it's un-register itself from `GltfImport`
- Even if last step fails, the null-check is now changed to a Unity-Object check, which would simply create a new default defer agent

That's as clean and bullet prove as it can get.

hth
